cfme/common/provider.py

```diff
@@ -158,8 +174,35 @@ def _submit(self, cancel, submit_button):
             submit_button()
             flash.assert_no_errors()
 
+    @variable(alias='rest')
     def create(self, cancel=False, validate_credentials=False):
         """
+        Creates a provider via the REST API
+        """
+        if cancel:
+            return
+        creds = []
+        for k, cred in self.credentials.iteritems():
+            if not cred:
+                logger.info('The credential {} has no type'.format(k))
+                continue
+            c_type = cred.type or 'default'
+            if cred.domain:
+                principal = r'{}\{}'.format(cred.domain, cred.principal)
+            else:
+                principal = cred.principal
+            creds.append({'auth_type': credential_type_map[c_type],
+                          'userid': principal,
+                          'password': cred.secret})
+
```

Comment:
We need "auth_key"  in creds which will include container token
https://github.com/ManageIQ/manageiq_docs/blob/master/api/appendices/resource_attributes.adoc